Q1 A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [ ]:
 import pandas as pd
 import numpy as np
 import scipy.stats as stats

In [ ]:
#load data
df=pd.read_csv("/content/Cutlets.csv")

In [ ]:
df.head(4)

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878

In [ ]:
unitA=pd.Series(df.iloc[:,0])
unitA.head(2)

0    6.8090
1    6.4376
Name: Unit A, dtype: float64

In [ ]:
unitB=pd.Series(df.iloc[:,1])

In [ ]:
p_value=stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [ ]:
p_value=stats.ttest_ind(df.iloc[:,0],df.iloc[:,1])
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

**conl:** As comparince to p value and α value = 0.05
 **p value > α** it's 0.47223 > 0.05

thus we can say, Null hyposthes is as Ho: μ1 = μ2 (There is no difference in diameters of cutlets between two units)

Thus Alternate hypothesis as Ha: μ1 ≠ μ2 (There is significant difference in diameters of cutlets between two units) 2 Sample 2 Tail test applicable

**Next question**



**Qn 2** A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.

 Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.


In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [ ]:
#dataset
df2=pd.read_csv('/content/LabTAT.csv')

In [ ]:
df2.head(3)

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18

In [ ]:
#anova
p_value=stats.f_oneway(df2.iloc[:,0],df2.iloc[:,1],df2.iloc[:,2],df2.iloc[:,3])

In [ ]:
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

**conl:** As comparince to p value and α value = 0.05
 **p value > α** it's 2.1156...14e-57 < 0.05

thus we can say, Null hyposthes is as Ho: l1≠l2≠l3≠l4

Thus,Accept Ha, hence Average of atleast 1 laboratory are different As per results we can say that these are not equal i.e Average of atleast 1 laboratory are different

**Next question**

Q 3 Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [ ]:
df3=pd.read_csv('/content/BuyerRatio.csv')
df3.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
df_table=df3.iloc[:,1:]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [ ]:
df_table.values 

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [ ]:
ch=stats.chi2_contingency(df_table)

In [ ]:
ch

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [ ]:
p_value=ch[1] # p value
p_value

0.6603094907091882

**conl:** As comparince to p value and α value = 0.05
 **p value > α** it's 0.660309 > 0.05

thus we can say, Null hyposthes is as Ho: man=woman

Thus,Accept H0, hence the  male and female buyers are similar across regions

**Next question**

QN 4   TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [ ]:
df4=pd.read_csv('/content/Costomer+OrderForm.csv')
df4

In [ ]:
#ways you can get 

In [ ]:
phi=df4.iloc[:,0]
phi

0      Error Free
1      Error Free
2      Error Free
3      Error Free
4      Error Free
          ...    
295    Error Free
296    Error Free
297    Error Free
298    Error Free
299    Error Free
Name: Phillippines, Length: 300, dtype: object

In [ ]:
df4.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
df4.groupby('Phillippines').size()

Phillippines
Defective      29
Error Free    271
dtype: int64

In [ ]:
phi=df4.iloc[:,0].value_counts()
phi

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
print(df4.iloc[:,0].value_counts(),df4.iloc[:,1].value_counts(),df4.iloc[:,2].value_counts(),df4.iloc[:,3].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [ ]:
arr=[df4.iloc[:,0].value_counts(),df4.iloc[:,1].value_counts(),df4.iloc[:,2].value_counts(),df4.iloc[:,3].value_counts()]

In [ ]:
type(arr)

list

In [ ]:
arr

[Error Free    271
 Defective      29
 Name: Phillippines, dtype: int64, Error Free    267
 Defective      33
 Name: Indonesia, dtype: int64, Error Free    269
 Defective      31
 Name: Malta, dtype: int64, Error Free    280
 Defective      20
 Name: India, dtype: int64]

In [ ]:
data_array=np.array([series.values for series in arr])
data_array

array([[271,  29],
       [267,  33],
       [269,  31],
       [280,  20]])

In [ ]:
print(data_array)

[[271  29]
 [267  33]
 [269  31]
 [280  20]]


In [ ]:
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [ ]:
chi=stats.chi2_contingency(obs)
chi

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [ ]:
p_value=chi[1]   #p value
p_value

0.2771020991233135

**conl:** As comparince to p value and α value = 0.05 p value > α it's 0.27710 > 0.05

Thus,Accept H0, hence customer order forms defective % does not varies by centre.